client/state/sites/vouchers/actions.js

```diff
@@ -75,6 +79,68 @@ export const vouchersRequestFailureAction = ( siteId, error ) => {
 	};
 };
 
+export const vouchersAssignReceiveAction = ( siteId, serviceType, voucher ) => {
+	return {
+		type: SITE_VOUCHERS_ASSIGN_RECEIVE,
+		siteId,
+		serviceType,
+		voucher
+	};
+};
+
+/**
+ * Action creator function
+ *
+ * Return SITE_VOUCHERS_ASSIGN_REQUEST action object
+ *
+ * @param {Number} siteId - side identifier
+ * @param {String} serviceType - service type
+ * @return {Object} siteId - action object
+ */
+export const vouchersAssignRequestAction = ( siteId, serviceType ) => {
+	return {
+		type: SITE_VOUCHERS_ASSIGN_REQUEST,
+		siteId,
+		serviceType
+	};
+};
+
+/**
+ * Action creator function
+ *
+ * Return SITE_VOUCHERS_ASSIGN_REQUEST_SUCCESS action object
+ *
+ * @param {Number} siteId - side identifier
+ * @param {String} serviceType - service type
+ * @return {Object} siteId - action object
+ */
+export const vouchersAssignRequestSuccessAction = ( siteId, serviceType ) => {
+	return {
+		type: SITE_VOUCHERS_ASSIGN_REQUEST_SUCCESS,
+		siteId,
+		serviceType
+	};
+};
+
+/**
+ * Action creator function
+ *
+ * Return SITE_VOUCHERS_ASSIGN_REQUEST_FAILURE action object
+ *
+ * @param {Number} siteId - site identifier
+ * @param {String} serviceType - service type
+ * @param {Object} error - error message according to REST-API error response
+ * @return {Object} action object
+ */
+export const vouchersAssignRequestFailureAction = ( siteId, serviceType, error ) => {
```

Comment:
yeah, it was more because it seems like we'll tend to end up with things like `vouchers.vouchersAssignRequestFailureAction()`

more of a question than a demand